cmd/metal-api/internal/metal/machine.go

```diff
@@ -165,6 +165,41 @@ type MachineNetwork struct {
 	Shared              bool     `rethinkdb:"shared" json:"shared"`
 }
 
+// NetworkType represents the type of a network
+type NetworkType string
+
+const (
+	PrivatePrimaryUnshared NetworkType = "privateprimaryunshared"
+	PrivatePrimaryShared   NetworkType = "privateprimaryshared"
+	PrivateSecondaryShared NetworkType = "privatesecondaryshared"
+	Underlay               NetworkType = "underlay"
+)
+
+// NetworkType determines the network type based on the flags stored in the db entity.
+func (mn *MachineNetwork) NetworkType() (*NetworkType, error) {
+	if mn.Underlay && !mn.PrivatePrimary && !mn.Private && !mn.Shared {
```

Comment:
if underlay is set to true i would set networktyp to underlay regardles of other flags ?

Reply:
if other flags contradict the underlay flag I would propose to raise an error.